### ISSUE TITLE
PathPlannerLib: add getter and setter for PathPlannerState curveRadius and deltaPos

### ISSUE THREAD

**Issue by dave-billin** [mjansen4857/pathplanner#296]
### TLDR
Please add setter and getter methods to the `PathPlannerState` class for accessing the `curveRadius` and `deltaPos` attributes.

### Background
I'm interested in creating a function similar to `transformStateForAlliance()`, but with the ability to apply arbitrary transformations to path states.  To do this, I'd like to copy and transform individual `PathPlannerState` objects returned by `PathPlannerTrajectory::getStates()`.  The `PathPlannerState` class has `public` scope.  But, there is presently no access provided to the `curveRadius` and `deltaPos` attributes, making it impossible to do a deep copy of those objects.  Minimally, I'd like it if a getter and setter were provided for those attributes.  If PathPlannerLib needs to keep those fields inaccessible, then perhaps providing a copy constructor or `clone()` function might be a suitable compromise that would allow deep-copy operations of PathPlannerTrajectory states.

My goal at the moment is to create a PathPlannerTrajectory transformation similar to the one provided by `transformTrajectoryForAlliance`, with the difference that it will keep the field coordinate system origin at the bottom side of the Blue Alliance.  I'm aware that facilities presently exist for moving the field coordinate system origin to the present alliance side.  But, my team has a body of code written around a single-origin paradigm that I'd like to preserve.  If I can copy `PathPlannerState` objects, making the translation is straightforward.  But, if I can't, the workaround will be more messy.

**Comment by mjansen4857** [comment]
This has been discussed in other issues before but I can't seem to find it to link it. Those values are not used at all after generation so they are not needed, which is why they are private. Your custom transformations can just ignore those and everything will be fine.
